### PR TITLE
Add vaccination information

### PR DESCRIPTION
**Pull request by divyagar**
Co-authored-by: Amit<amitkr0819@gmail.com>
Co-authored-by: Khavin<khavinshankar@gmail.com>

#945

### CODE REVIEW

File: src/Components/Patient/PatientRegister.tsx [hunk 1]
```diff
@@ -147,6 +147,8 @@ const initForm: any = {
   cluster_name: "",
   covin_id: "",
   is_vaccinated: "false",
+  number_of_doses: 0,
+  vaccine_name: "CoviShield",
```
Comment (dauntlessnomad) [inline src/Components/Patient/PatientRegister.tsx]: are we considering only covishield ?

Comment (dauntlessnomad) [reply src/Components/Patient/PatientRegister.tsx]: @divyagar

Comment (khavinshankar) [inline src/Components/Patient/PatientRegister.tsx]: make it `vaccine_name: "",` or `vaccine_name: null,`

File: src/Components/Patient/PatientRegister.tsx [hunk 2]
```diff
@@ -278,6 +280,8 @@ export const PatientRegister = (props: PatientRegisterProps) => {
     }
   };
 
+  const vaccineNames = ["CoviShield", "Covaxin"];
```
Comment (khavinshankar) [inline src/Components/Patient/PatientRegister.tsx]: add Select to the array, const vaccineNames = ["Select", "CoviShield", "Covaxin"];

File: src/Components/Patient/PatientRegister.tsx [hunk 3]
```diff
@@ -595,6 +599,8 @@ export const PatientRegister = (props: PatientRegisterProps) => {
         srf_id: state.form.srf_id,
         covin_id: state.form.covin_id,
         is_vaccinated: state.form.is_vaccinated,
+        number_of_doses: Number(state.form.number_of_doses),
+        vaccine_name: state.form.vaccine_name,
```
Comment (khavinshankar) [inline src/Components/Patient/PatientRegister.tsx]: add a check to see if the value is _Select_, `vaccine_name: state.form.vaccine_name && state.form.vaccine_name !== 'Select' ? state.form.vaccine_name : undefined,`

**Review by khavinshankar — COMMENTED** [review]
You are having Covishield as the default vaccine instead, try this. This is how it is implemented in the rest of the components, just for consistency...

File: src/Components/Patient/PatientRegister.tsx (hunk 1 above)
Comment (divyagar) [reply src/Components/Patient/PatientRegister.tsx]: @tomahawk-pilot  No this is the default selected value.

File: src/Components/Patient/PatientRegister.tsx (hunk 3 above)
Comment (divyagar) [reply src/Components/Patient/PatientRegister.tsx]: @khavinshankar  Thanks. I have made changes.

File: src/Components/Patient/PatientRegister.tsx [hunk 4]
```diff
@@ -278,6 +280,8 @@ export const PatientRegister = (props: PatientRegisterProps) => {
     }
   };
 
+  const vaccineNames = ["Select", "CoviShield", "Covaxin"];
```
Comment (bodhish) [inline src/Components/Patient/PatientRegister.tsx]: Move this to constants

**Comment by dauntlessnomad** [comment]
@divyagar and @khavinshankar  ,
please dont format the code beyond a point where its impossible to review the changes.

**Comment by shaileshaanand** [comment]
#1050 fixes the linting issues, please see #1050 @bodhish @vigneshhari @amruthavijay88

**Pull request closed without merging**